### DISCUSSION TITLE
Add a custom oauth2 provider with JS hooks

### DISCUSSION THREAD

**Discussion by ThomasSelvig** [pocketbase/pocketbase#3427] [Q&A]
Is it possible to use hooks to add Bungie as an oauth2 provider? What would code for adding any provider look like?
This is how they describe their protocol: https://github.com/Bungie-net/api/wiki/OAuth-Documentation

**Comment by ganigeorgiev** [comment]
No, it is not possible at the moment to register dynamically an external OAuth2 provider.

Depending on the users demand and whether the provider is generic enough, eventually it can be integrated directly into the core package (see for example the recent [mailcow PR](https://github.com/pocketbase/pocketbase/pull/3364)), but I'm not familiar with Bungie and whether it makes sense to be used as a generic auth provider.

**Comment by ThomasSelvig** [comment]
It's not generic enough to take that high of a priority over other providers, so it depends on if you consider more providers to be worth the "bloat". It's generic enough to be on this (bit opinionated) list: https://next-auth.js.org/v3/configuration/providers#available-providers
